### ISSUE TITLE
Button data is binary, but is treated as if it were continuous

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#3398]
## What problem does this solve or what need does it fill?

Button data cannot be analogue, it is etiher exactly 0.0 or 1.0 (at least, when returned by `gilrs`). See the investigation in https://github.com/bevyengine/bevy/issues/3246#issuecomment-997977629.

However, we return a `f32`, which is always (at least, when using `gilrs`) either exactly 0.0 or 1.0. 

Moreover, we have an entirely useless `ButtonSettings` struct, which thresholds these returned values in a "configurable" way, and converts them back into a bool 🤦🏽‍♀️

## What solution would you like?

1. Eliminate `ButtonSettings` completely.
2. Only expose a binary value to end-users.
3. Store the received button input value as a `bool` as soon as we receive it from `gilrs`.
4. Further investigate why we're receiving a `f32` from `gilrs` in the first place.
5. Work with the `gilrs` team as needed to get this data to us as a `bool` and skip the float conversion as needed.

I would probably prefer a 
```rust
enum ButtonState {
  Pressed,
  NotPressed
}
```
as our representation oif this data, but even a `bool` would be significantly better than our current `f32`.

## What alternative(s) have you considered?

We could continue to allow a `f32` value for button input, in the hopes of eventually supporting a generic "analogue button" solution.
This seems impractical to me, because:

1. `gilrs`, the dominant input library for this need does not handle analogue buttons at all.
2. We waste work on every button press performing several conversions and comparisions.
3. Buttons are, nearly by definition, binary. If new hardware emerges that needs this paradigm, we will need to support it directly.
4. If people want to experiment with strange hardware and analogue buttons, the `Axis` type already exists.

## Additional context

Identified in #3246.

**Comment by alice-i-cecile** [comment]
If you're tackling this as your first issue, feel free to stop after step 3 :) Step 4 and 5 can be comfortably split out, just open a new issue when your PR is ready to track that we could clean this up further.

**Comment by alice-i-cecile** [comment]
Interestingly, the web standard uses a [0.0, 1.0] range as well, and seems to suggest that analogue inputs could exist: https://developer.mozilla.org/en-US/docs/Web/API/GamepadButton/value 

See https://doc.stride3d.net/4.0/en/manual/input/gamepads.html.

I think I see the distinction here: 
- Bevy seems to treat analogue buttons like triggers as axes, rather than standard buttons.
- https://w3c.github.io/gamepad/#dom-gamepadbutton-value treats triggers as buttons, and discretizes them.

**Comment by cart** [comment]
Given that for a given input name, it might be analog or binary depending on the controller, I think it makes sense for us to allow inputs to be treated as either an axis _or_ a button, depending on the needs of the game.

Analog axes can be treated as binary buttons by defining a "pressed range". Binary buttons can be treated as axes by mapping "not pressed" to 0.0 and "pressed" to 1.0.

**Comment by alice-i-cecile** [comment]
> Given that for a given input name, it might be analog or binary depending on the controller, I think it makes sense for us to allow inputs to be treated as either an axis or a button, depending on the needs of the game.

I agree: different controllers and different games might want to treat the same button as either binary or analog.

IMO the correct way to do this is to expose a `value` method on the button, that specific games can access. This allows us to follow the specific, and clearly differentiate between Axes ([-1.0, 1.0]) and Buttons ([0.0, 1.0]) in a principled way.
